hca_bundle_tools/file_metadata_to_csv.py

```diff
@@ -149,6 +150,14 @@ def add_bundle_files_to_row(self, list_of_metadata_objects, dir_name=None):
             obj["file_name"] = self._deep_get(content, ["file_core", "file_name"])
             obj["file_format"] = self._deep_get(content, ["file_core", "file_format"])
 
+            file_segments = obj["file_name"].split('.')
+
+            if len(file_segments) > 1 and file_segments[-1] in self.default_blocked_file_ext:
+                continue
+
+            if '.zarr!' in obj["file_name"] and not '.zarr!.zattrs' in obj['file_name']:
```

Comment:
Can you explain why you do this here for zarr files? I don't know much about this file type.

Reply:
https://zarr.readthedocs.io/en/stable/spec/v2.html

zarr stores are directories—the `!` in the file path is the result of a hack in analysis that substitutes a slash with an exclamation point—and can contain hundreds or thousands of files. The current matrix zarr stores contain dozens. It's pointless to include a line for each in the TSV because the other metatdata columns will have the same values for all files in the store. 

I've talked to @GPelayo about this and he will modify the code to write a single TSV row representing an entire zarr store.

The DSS has added support for directories in bundles which will make the `!` hack unnecessary. @GPelayo will add code that gracefully handles both `!` and `/` in file paths for zarr member files.